### PR TITLE
Add an option to customize shfmt behaviour

### PR DESCRIPTION
**Pull request by jovandeginste**
Signed-off-by: Jo Vandeginste <Jo.Vandeginste@kuleuven.be>

### CODE REVIEW

**Comment by jovandeginste** [comment]
The build failed with an unclear error - should I trigger a rebuild or something?

File: autoload/neoformat/formatters/sh.vim [hunk 1]
```diff
@@ -3,9 +3,11 @@ function! neoformat#formatters#sh#enabled() abort
 endfunction
 
 function! neoformat#formatters#sh#shfmt() abort
+    let opts = get(g:, 'shfmt_opt', '')
+
     return {
             \ 'exe': 'shfmt',
-            \ 'args': ['-i ' . shiftwidth()],
+            \ 'args': ['-i ' . shiftwidth(), opts],
```
Comment (sbdchd) [inline autoload/neoformat/formatters/sh.vim]: should we add a space `' '` between the `shiftwidth()` and `opts`

**Comment by sbdchd** [comment]
CI's pretty broken on this project so I wouldn't worry about it

File: autoload/neoformat/formatters/sh.vim (hunk 1 above)
Comment (jovandeginste) [reply autoload/neoformat/formatters/sh.vim]: I was assuming that was the behaviour of the array. Am I wrong? It seemd to work as expected in my tests...

Comment (jovandeginste) [reply autoload/neoformat/formatters/sh.vim]: That being said: if that's the case, we could rewrite that to `['-i', shiftwidth(), opts]`

Comment (sbdchd) [reply autoload/neoformat/formatters/sh.vim]: Oh I think you're right

**Comment by jovandeginste** [comment]
thanks for merging! (and your project in general)

**Pull request merged**